py/server/tests/test_parquet.py

```diff
@@ -576,6 +581,85 @@ def test_read_parquet_from_s3(self):
             read("s3://dh-s3-parquet-test1/multiColFile.parquet", special_instructions=s3_instructions).select()
         # TODO(deephaven-core#5064): Add support for local S3 testing
 
+    def verify_metadata_files(self, root_dir):
+        metadata_file_path = os.path.join(root_dir, '_metadata')
+        self.assertTrue(os.path.exists(metadata_file_path))
+        common_metadata_file_path = os.path.join(root_dir, '_common_metadata')
+        self.assertTrue(os.path.exists(common_metadata_file_path))
+
+    def test_writing_partitioned_data(self):
+        source = new_table([
+            string_col("X", ["Aa", "Bb", "Aa", "Cc", "Bb", "Aa", "Bb", "Bb", "Cc"]),
+            string_col("Y", ["M", "N", "O", "N", "P", "M", "O", "P", "M"]),
+            int_col("Number", [55, 76, 20, 130, 230, 50, 73, 137, 214]),
+        ])
+        partitioned_table = source.partition_by(by="X")
+        base_name = "test-{uuid}"
+        max_dictionary_keys = 1024
+
+        root_dir = os.path.join(self.temp_dir.name, "test_partitioned_writing")
+        if os.path.exists(root_dir):
+            shutil.rmtree(root_dir)
+
+        def verify_table_from_disk(table):
+            self.assertTrue(len(table.columns))
+            self.assertTrue(table.columns[0].name == "X")
+            self.assertTrue(table.columns[0].column_type == ColumnType.PARTITIONING)
+            self.assert_table_equals(table.select().sort("X", "Y"), source.sort("X", "Y"))
+
+        def verify_file_names():
+            partition_dir_path = os.path.join(root_dir, 'X=Aa')
+            self.assertTrue(os.path.exists(partition_dir_path))
+            self.assertTrue(len(os.listdir(partition_dir_path)) == 1)
+            parquet_file = os.listdir(partition_dir_path)[0]
+            self.assertTrue(fnmatch.fnmatch(parquet_file, 'test-*.parquet'))
+
+        # Test all different APIs
+        write_key_value_partitioned_table(partitioned_table, destination_dir=root_dir, base_name=base_name,
+                                          max_dictionary_keys=max_dictionary_keys)
+        from_disk = read(root_dir)
+        definition = from_disk.columns
+        verify_table_from_disk(from_disk)
+        verify_file_names()
+        from_disk = None
+
+        shutil.rmtree(root_dir)
```

Comment:
If you need to make certain these root_dirs get cleaned up in case there is a problem and the test is rerun, consider try-finally or a shutdown for the tests or putting a rmtree before the first test.
https://www.geeksforgeeks.org/try-except-else-and-finally-in-python/

Reply:
No, I don't want to rerun any tests. The test can fail the first time it fails.